code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,138 @@
+     * Creates an instance of [AepButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = map[MessagingConstants.AepUIKeys.INTERACT_ID] as String,
+            actionUrl = map[MessagingConstants.AepUIKeys.ACTION_URL] as String,
+            text = (map[MessagingConstants.AepUIKeys.TEXT] as Map<String, Any>).let { createAepText(it) }
+        )
+    }
+
+    /**
+     * Creates an instance of [AepDismissButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    fun createAepDismissButton(map: Map<String, Any>): AepDismissButton {
+        return AepDismissButton(
+            style = map[MessagingConstants.AepUIKeys.STYLE] as? String ?: "NONE_ICON"
+        )
+    }
+}
+
+/**
+ * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+ *
+ * @param contentCardSchemaData The content card schema data containing the content map.
+ * @return The parsed [AepUITemplate] or null if parsing fails.
+ * @throws IllegalArgumentException If the content map is null or if there is an error during parsing.
+ */
+fun getTemplateModelFromContentCardSchemaData(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+    val contentMap =
+        contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+    try {
+        val id = contentMap[MessagingConstants.AepUIKeys.ID] as String
+        val title =
+            ContentCardSchemaDataUtils.createAepText(contentMap[MessagingConstants.AepUIKeys.TITLE] as Map<String, Any>)
+        val body = (contentMap[MessagingConstants.AepUIKeys.BODY] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepText(it)
+        }
+        val image = (contentMap[MessagingConstants.AepUIKeys.IMAGE] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepImage(it)
+        }
+        val actionUrl = contentMap[MessagingConstants.AepUIKeys.ACTION_URL] as? String
+        val buttons = (contentMap[MessagingConstants.AepUIKeys.BUTTONS] as? List<Map<String, Any>>)?.map {
+            ContentCardSchemaDataUtils.createAepButton(it)
+        }
+        val dismissBtn = (contentMap[MessagingConstants.AepUIKeys.DISMISS_BTN] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepDismissButton(it)
+        }
+        return SmallImageTemplate(
+            id = id,
+            title = title,
+            body = body,
+            image = image,
+            actionUrl = actionUrl,
+            buttons = buttons,
+            dismissBtn = dismissBtn
+        )
+    } catch (e: Exception) {
+        throw IllegalArgumentException("Error parsing content map: ${e.message}")
```

Comment:
If we use DataReader.opt* we can avoid throwing this exception

Reply:
Returned null in case of parsing fails.